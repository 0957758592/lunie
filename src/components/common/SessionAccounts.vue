<template>
  <SessionFrame>
    <div class="session">
      <div class="session-container">
        <h2 class="session-title">My accounts</h2>
        <div class="session-main">
          <p>
            You can use the accounts below to explore
            <router-link to="/wallet">
              Lunie
            </router-link>
            and to approve transactions.
          </p>

          <AccountList :accounts="accounts" />

          <div class="button-add-account">
            <TmBtn
              value="Add Another Account"
              color="primary"
              @click.native="addAccount"
            />
          </div>
        </div>
      </div>
    </div>
  </SessionFrame>
</template>

<script>
import AccountList from "common/AccountList"
import SessionFrame from "common/SessionFrame"
import TmBtn from "common/TmBtn"
export default {
  name: `session-accounts`,
  components: {
    AccountList,
    SessionFrame,
    TmBtn
  },
  computed: {
    accounts() {
      let accounts = this.$store.state.accounts
      return accounts.map(({ name, address }) => ({
        address,
        name
      }))
    }
  },
  methods: {
    addAccount() {
      this.$router.push(`/welcome`)
    }
  }
}
</script>

<style scoped>
.button-add-account {
  display: flex;
  justify-content: flex-end;
  padding: 1.5rem 0 1rem;

  /* keeps button in bottom right no matter the size of the action modal */
  flex-grow: 1;
  align-self: flex-end;
}
</style>
