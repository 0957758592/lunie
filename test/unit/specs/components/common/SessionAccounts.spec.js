import { createLocalVue, shallowMount } from "@vue/test-utils"
import Vuelidate from "vuelidate"
import SessionAccounts from "common/SessionAccounts"

describe(`SessionAccounts`, () => {
  const localVue = createLocalVue()
  localVue.use(Vuelidate)

  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(SessionAccounts, {
      localVue,
      stubs: [`router-link`]
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`should close`, () => {
    const self = {
      $emit: jest.fn()
    }
    SessionAccounts.methods.close.call(self)
    expect(self.$emit).toHaveBeenCalledWith(`close`)
  })

  it("moves to other session pages", () => {
    const self = {
      $emit: jest.fn()
    }
    SessionAccounts.methods.setState.call(self, "welcome")
    expect(self.$emit).toHaveBeenCalledWith("route-change", "welcome")
  })
})
