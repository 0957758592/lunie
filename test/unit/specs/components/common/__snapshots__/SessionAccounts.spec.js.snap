// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionAccounts has the expected html structure 1`] = `
<div
  class="session-frame"
  mode="out-in"
  name="component-fade"
>
  <router-link-stub
    to="/"
  >
    <img
      class="session-logo"
      src="~assets/images/cosmos-wallet-logo.svg"
    />
  </router-link-stub>
   
  <div
    class="session-outer-container"
  >
    <div
      class="session"
    >
      <div
        class="session-container"
      >
        <h2
          class="session-title"
        >
          My accounts
        </h2>
         
        <div
          class="session-main"
        >
          <p>
            
          You can use the accounts below to explore
          
            <router-link-stub
              to="/wallet"
            >
              
            Lunie
          
            </router-link-stub>
            
          and to approve transactions.
        
          </p>
           
          <ul
            class="account-list"
          >
            <li
              class="account"
            >
              <div
                class="account-info"
              >
                <h3>
                  Benjis account
                </h3>
                 
                <div
                  class="bech32-address"
                >
                  <div
                    class="address"
                    id="address"
                  >
                    
    cosmosâ€¦4v9e
  
                  </div>
                   
                  <div
                    class="copied"
                  >
                    <i
                      class="material-icons"
                    >
                      check
                    </i>
                    <span>
                      Copied
                    </span>
                  </div>
                </div>
              </div>
               
              <button
                class="tm-btn account-button"
              >
                <span
                  class="tm-btn__container tm-btn--primary"
                >
                  <!---->
                  <span
                    class="tm-btn__value"
                  >
                    Use Account
                  </span>
                </span>
              </button>
            </li>
          </ul>
           
          <div
            class="button-add-account"
          >
            <button
              class="tm-btn"
            >
              <span
                class="tm-btn__container tm-btn--primary"
              >
                <!---->
                <span
                  class="tm-btn__value"
                >
                  Add Another Account
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <button
    class="tm-btn session-close"
  >
    <span
      class="tm-btn__container tm-btn--secondary"
    >
      <!---->
      <span
        class="tm-btn__value"
      >
        Back to Lunie
      </span>
    </span>
  </button>
</div>
`;
